webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js

```diff
@@ -0,0 +1,60 @@
+const buildTableRow = (subscription, availableQuantities, updatedQuantity) => {
+  if (updatedQuantity[subscription.id]) {
+    return {
+      ...subscription,
+      entitlementsChanged: true,
+      quantity: updatedQuantity[subscription.id],
+      availableQuantity: availableQuantities[subscription.id],
+    };
+  }
+  return {
+    ...subscription,
+    availableQuantity: availableQuantities[subscription.id],
+  };
+};
+
+const buildTableRowsFromGroup = (subscriptionGroup, availableQuantities, updatedQuantity) => {
+  const { open, subscriptions } = subscriptionGroup;
+
+  // build row for each subscription
+  if (open) {
+    return subscriptions.map(subscription =>
+      buildTableRow(subscription, availableQuantities, updatedQuantity));
+  }
+
+  // build row only for the first subscription in the group
+  const [subscription] = subscriptions;
+  return [buildTableRow(subscription, availableQuantities, updatedQuantity)];
+};
+
+export const buildTableRows = (groupdSubscriptions, availableQuantities, updatedQuantity) => {
+  const rows = [];
+
+  Object.values(groupdSubscriptions).forEach(subscriptionGroup =>
```

Comment:
This would invoked each render, even without any change

Reply:
> Yes, the function run before each render, but it will run the inner logic only if the subscriptions received from redux changed.
>
> Once we will move all the state to redux, we would be able to implement memorized selectors.